### ISSUE TITLE
Proposal: Transform assets.

### ISSUE THREAD

**Issue by IngwiePhoenix** [webpack/webpack#1621]
We have all the logic to translate/transform JavaScript files.

But why, in the world, can't we just transform an asset?

For instance: I am using a EJS template to render error pages for when my web server really has no other option left. So for that, I made a tiny SCSS. But, unfortunately, it won't quite work that way.

I now have an almost empty JS file and my transformed SCSS file, since i had the text-extract plugin. But is this _really_ the right way?

I say no.

In addition to `entry`, I want to suggest `assets`. For example:

``` javascript
module.exports = {
    // Scripts
    entry: {
        libwebpack: [/* ... */],
        main: "BIRD3/App/Entry/Browser/Main",
        compat: "BIRD3/Support/Compatibility"
    },
    // Assets
    assets: {
        error: "BIRD3/Frontend/Design/Style/error.scss"
    },
    // ... snip ...
}
```

Essentially, `assets` would be transformed like usual, except that they would be emitted directly. To further enhance behaviour, we would possibly want to make a new loader option available:

``` javascript
if(this.isAssetResource) {
    // Do not generate JS code like usual. I.E. just turn SCSS to CSS, SVG sprites to a big one, ...
} else {
    // Generate JS.
}
```
- `isAssetResource`: True, if this resource should be treated as an asset, which is not directly related to the main entries.

+1 this if you agree. Thanks! :)

**Comment by jenius** [comment]
+1

**Comment by sokra** [comment]
I would like to avoid this as this will lead to the wrong style. People would start keeping CSS and JS separate (as with grunt or gulp). Instead webpack should encourage people to keep CSS and JS together in a single dependency tree.

**Comment by IngwiePhoenix** [comment]
But what about scenarios, when the JavaScript is basically 0 bytes?

In my scenario, I have an error page for which I wrote some SCSS. I’d like to render the SCSS through WebPack’s loaders and get the resulting CSS. 

Of course, I can just specify it as an entry point, which I did, and grab the CSS. But there is an unneccessary js file being laid out thats not even going to be used.

What solution(s) is/are there?
